packages/flutter_tools/lib/src/ios/cocoapods.dart

```diff
@@ -0,0 +1,123 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import '../base/common.dart';
+import '../base/context.dart';
+import '../base/file_system.dart';
+import '../base/io.dart';
+import '../base/logger.dart';
+import '../base/process.dart';
+import '../base/process_manager.dart';
+import '../base/version.dart';
+import '../cache.dart';
+import '../globals.dart';
+
+final String noCocoaPodsConsequence = '''
+  CocoaPods is used to retrieve the iOS platform side's plugin code that responds to your plugin usage on the Dart side.
+  Without resolving iOS dependencies with CocoaPods, plugins will not work on iOS.
+  For more info, see https://flutter.io/platform-plugins''';
+
+final String cocoaPodsInstallInstructions = '''
+  brew update
+  brew install cocoapods
+  pod setup''';
+
+final String cocoaPodsUpgradeInstructions = '''
+  brew update
+  brew upgrade cocoapods
+  pod setup''';
+
+CocoaPods get cocoaPods => context.putIfAbsent(CocoaPods, () => const CocoaPods());
+
+class CocoaPods {
+  const CocoaPods();
+
+  Future<bool> get hasCocoaPods => exitsHappyAsync(<String>['pod', '--version']);
+
+  String get cocoaPodsMinimumVersion => '1.0.0';
+
+  Future<String> get cocoaPodsVersionText async => (await runAsync(<String>['pod', '--version'])).processResult.stdout.trim();
+
+  Future<bool> get isCocoaPodsInstalledAndMeetsVersionCheck async {
+    if (!await hasCocoaPods)
+      return false;
+    try {
+      final Version installedVersion = new Version.parse(await cocoaPodsVersionText);
+      return installedVersion >= new Version.parse(cocoaPodsMinimumVersion);
+    } on FormatException {
+      return false;
+    }
+  }
+
+  /// Whether CocoaPods ran 'pod setup' once where the costly pods' specs are cloned.
+  Future<bool> get isCocoaPodsInitialized => fs.isDirectory(fs.path.join(homeDirPath, '.cocoapods', 'repos', 'master'));
+
+  Future<Null> processPods(Directory appIosDir, String iosEngineDir) async {
+    if (await _checkPodCondition()) {
```

Comment:
Since we pretty much know we need Pods if we have plugins, shouldn't we now fail the build here if the project uses plugins but CocoaPods isn't installed?